imapclient/imapclient.py

```diff
@@ -192,17 +192,20 @@ def starttls(self, ssl_context=None):
         Raises :py:exc:`AbortError` if the server does not support STARTTLS
         or an SSL connection is already established.
         """
-        if self.ssl or self._starttls_done:
-            raise self.AbortError('TLS session already established')
+        try:
+            return self._imap.starttls(ssl_context=ssl_context)
+        except AttributeError:
+            if self.ssl or self._starttls_done:
```

Comment:
It's a little clearer IMO if the except clause was just a `pass` with the `if self.ssl ...` bit at the function level.